### DISCUSSION TITLE
Custom error messages

### DISCUSSION THREAD

**Discussion by notmeta** [ory/kratos#1715] [Q&A] [answered]
Apologies if this is mentioned anywhere on the docs, I've had a look but been unable to find anything specifically.

Is there any way to override the error messages returned in flows?
i.e. I only have email/password login enabled, but if an invalid password is given it mentions the identifier could be a phone number, username, etc. which could be misleading.

Is it up to the implementation to override error messages (based on error id, etc), or is there a built-in way to do so? 

Thanks

**Comment by aeneasr** [comment, marked as answer]
You would need to do that in the UI that you write. All text messages have a unique identifier which you can use to e.g. translate or customize the message. Text messages also have a context with variables that you might need. See: https://www.ory.sh/kratos/docs/concepts/ui-user-interface

In Ory Kratos itself, it is not possible to customize this and never will be!

**Reply by dro-sh** [reply]
@aeneasr There is a way to set in context key for i18n? For example I have a **Password** input or **Sign In** button. Can I set in context for this fields key? Or in my UI I should use ID of fields and use ID as a key in my locales json files? If I can how is it configured?

P.S.
I found [getNodeId](https://github.com/ory/integrations/blob/main/src/ui/index.ts#L98), which return ID for nodes. But for buttons, like `Sign In` and `Sign Up` has ID `method`. Is there a way to override this? I can use different namespaces. No question more :)
